include/boost/geometry/strategies/geographic/azimuth.hpp

```diff
@@ -52,41 +52,52 @@ public :
         return m_spheroid;
     }
 
-    template <typename T>
+    template
+    <
+        bool EnableAzimuth,
+        bool EnableReverseAzimuth,
+        typename T
+    >
```

Comment:
As mentioned before. I'm not sure that this is a good function design because the user needs to have template arguments and function arguments synchronized. I'd rather implement this as functions taking exactly these output arguments which will be returned. In case of overloads returning single value it could be returned from the function instead of taken as output argument. I.e.:

    strategy.apply(..., azi1, azi2);
    T azi1 = strategy.apply(...);
    T azi2 = strategy.apply_reverse(...);